From ecd0c86d62351d267bdc9566286c532a394c711b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ingo=20Kl=C3=B6cker?= <dev@ingo-kloecker.de>
Date: Mon, 2 Sep 2024 15:33:25 +0200
Subject: [PATCH] build,python: Fix build with setuptools 72.2.0+

* lang/python/setup.py.in (BuildExtFirstHack.run): Extend members of the
extension instead of lists that were passed to the extension.
--

setuptools 72.2.0 integrated changes in distutils which included
"Support for Pathlike objects in data files and extensions". With this
change the extensions now take a copy of the sources list passed to the
constructor instead of keeping a reference to the passed list. Hence,
modifying the sources list that was passed to the extension didn't
change the sources list of the extension anymore. This is fixed by
modifying the sources list of the extension directly. For consistency
we do the same for the swig_opts list.

GnuPG-bug-id: 7281
---
 lang/python/setup.py.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lang/python/setup.py.in b/lang/python/setup.py.in
index 7b64ba7b..d6f537e0 100755
--- a/lang/python/setup.py.in
+++ b/lang/python/setup.py.in
@@ -217,9 +217,9 @@ class BuildExtFirstHack(build):
     def run(self):
         self._generate()
 
-        swig_sources.extend((self._in_build_base('gpgme.i'),
-                             self._in_build_base('helpers.c')))
-        swig_opts.extend([
+        swige.sources.extend((self._in_build_base('gpgme.i'),
+                              self._in_build_base('helpers.c')))
+        swige.swig_opts.extend([
             '-I' + self.build_base, '-outdir',
             os.path.join(self.build_lib, 'gpg')
         ])
